applications/spacewalk-bridge.md

```diff
@@ -0,0 +1,231 @@
+- **Bridge Pallet**: this is the main component of the Spacewalk bridge that implements all logic on the side of the Substrate-based chain. It is particularly responsible for minting tokens during deposits and burning tokens during withdrawals. The storage of the bridge pallet maintains the complete state that is required for the bridge to work correctly. This state contains (among others): the account ids of the Stellar bridge accounts, the association of bridge accounts to Stellar assets, book keeping information about the state of the Stellar network and assets locked in the bridge accounts. It also maintains the collateral of the bridge nodes and slashes it in case one of the bridge nodes misbehaves.
+- **Bridge Nodes**: these are privately owned nodes that watch changes in the storage of the bridge and create, sign and submit Stellar transactions that a) unlock tokens during a withdrawal and b) maintain the set of bridge accounts and its signers. Each bridge nodes contains some secret value. Together these secrets can be used to sign transactions for the bridge accounts. The bridge nodes use a t-out-of-n threshold signing scheme for the bridge accounts. For n ≤ 20 this is achieved through multisignature accounts that are directly supported through the Stellar protocol and for n > 20 this is achieved through key aggregation for Schnorr signatures (see [Maxwell, Gregory, et al. "Simple schnorr multi-signatures with applications to bitcoin"](https://eprint.iacr.org/2018/068.pdf)). Every bridge node needs to lock a certain amount of PEN tokens as a collateral in the bridge pallet. These tokens are slashed in case the bridge node misbehaves.
+- **Stellar Oracle**: is a system that provides information about the state of the Stellar network to the bridge pallet. In XCLAIM this is implemented through a chain relay in the bridge pallet. However, we will aim for a solution where every node of the Substrate-based chain is co-located with a Stellar validator node, which it uses as an oracle.
+
+**Out of scope**
+
+The following aspects are out of scope of the current proposal and subject to future applications:
+
+- Stellar protocol updates that are required to implement a chain relay/light client/simplified payment verification for Stellar as a Substrate pallet
```

Comment:
what protocol updates are required? hasn't [this](https://github.com/stellar/slingshot/pull/308) already walked a long stretch of the path?

Reply:
@Lederstrumpf You are referring to Stellar's project slingshot, which is a different architecture from Stellar. For example, slingshot uses [unspent outputs](https://github.com/stellar/slingshot/blob/main/zkvm/docs/zkvm-design.md#scalability) whereas Stellar does not. Utreexo is based on architectures that use unspent output but is not directly applicable to Stellar in that sense.

The main reason why Stellar does not allow for SPV is that Stellar blocks don't use Merkle trees. For that reason there are no efficient inclusion proofs (such as Merkle proofs) possible with the current protocol definition of Stellar.